### ISSUE TITLE
Mailchimp notifications bunkers

### ISSUE THREAD

**Issue by likewhoa** [NYC-Camp/website#236]
We have an issue with Mailchimp showing "You have updated your settings in NYC Camp 14 Email List." on every user profile page load, whether you're authenticated or not.

**Comment by esod** [comment]
For now we can disable the message. pull request to follow.

**Comment by esod** [comment]
https://github.com/NYC-Camp/website/pull/237

**Comment by esod** [comment]
mailchimp_update_user() is in only two places in the stack. Its function creation in mailchimp.module line 139 and its call in mailchimp_lists.module line 689.

**Comment by esod** [comment]
For now I commented out the message. We can keep this issue open to develop a more a cause related resolution. 

The ux benefits well from not printing the message.

**Comment by willykaram** [comment]
Agreed on improving the UX experience. But we should likely open an news issue to address debugging what's wrong with MailChimp.

**Comment by willykaram** [comment]
Guys we should easily be able to restore the user images will a script that runs against the DB. This should be able to be done without taking the site down. We can also run it on Dev/Test or a Multidev to confirm all is working smoothly and any QA is done.

**Comment by likewhoa** [comment]
@willykaram it's possible by just updating the missing value in 'picture' column in the users table from an old working db, probably one from before March 28th. I have already tested locally that restoring that value restores the original picture. What needs to be done is just do a compare and replace from old to new.

**Comment by willykaram** [comment]
@likewhoa why to we need to take the site down to do this? We can use a
mysql client or CLI with transaction queries to update the DB.

**Comment by likewhoa** [comment]
@willykaram lets bring this discussion to the appropriate issue queue, this thread is not on topic to user profile images and I only responded to you thinking you was on the relevant issue.

**Comment by mirie** [comment]
Ok, I'm suspecting a rogue Rule here. I just want to confirm that it was on every user profile view and not on all pages. 

The suspicious rule is: field_mobile_code /admin/config/workflow/rules/reaction/manage/rules_default_value_for_field_mobile_code_ 

because it has an action for Set a data value. I'm thinking that this is what is causing the user save/update cascade.

**Comment by ForestMars** [comment]
Is the rule in question under code control? 

More to the point, do we have m/any rules not under code control?

**Comment by mirie** [comment]
I have no idea. In fact, I do not even know our actual workflow for this :toilet:

**Comment by ForestMars** [comment]
The workflow is that rules should only be added if there is a related ticket (ie "we need x to do y")

Adding all rules in code is preferred. Any rule added in the ui should be pulled down with a database sync, featurized, and pushed back up in code. 

Also rules should make use of tokens, and not hard-coded values that apply to NYC Camp only.

**Comment by mirie** [comment]
ok @likewhoa, @esod, please take a look at: http://mailchimp-nyccamp2014.gotpantheon.com/user

I brought back that message and disabled that rule. 

However, that rule was most likely created for a reason. Do either you you know about this rule? I think we can set the value for field-mobile-code in a way that would not involve an db update for the user object. Thoughts?

**Comment by likewhoa** [comment]
that rule seems like something related to what @esod did w.r.t qrcodes but I haven't even looked at it just an assumption.

**Comment by esod** [comment]
The rule drives the QR code. It is in code. Why would "Set a data value" and "User account page is viewed" be in Rules if they weren't safe to use. How can standard Rules methods be causing the save/update cascade in mailchimp?

**Comment by mirie** [comment]
The "Set a data value" is really the issue, right? I'm not sure I am
understanding your question

**Comment by esod** [comment]
Why is "Set a data value" an issue? Why is the rule an issue? I think we're looking in the wrong place.

**Comment by likewhoa** [comment]
I don't think the 'set a data value' is the cause here guys. It's something else because if you go to a users profile, click edit, then save it you don't see the mailchimp msg. 

We need to dig deeper.

**Comment by mirie** [comment]
@likewhoa, are u checking this on this multidev branch?

**Comment by likewhoa** [comment]
on live @mirie

**Comment by mirie** [comment]
@likewhoa, you're not seeing the Mailchimp messages on master because that
line is currently commented out.

This is not the case on http://mailchimp-nyccamp2014.gotpantheon.com/user

Also, I created this separate environment to be able to play around with
the rule theory.

**Comment by esod** [comment]
Looks like with the qr_code module disabled, the message stops firing. I'll disable the module and we'll watch the logs for a while. Fooey.

**Comment by esod** [comment]
https://drupal.org/comment/8654077#comment-8654077

**Comment by likewhoa** [comment]
@esod assigning to you since qr-code is your baby.
